packages/x-date-pickers/src/MonthCalendar/PickersMonth.tsx

```diff
@@ -12,11 +12,14 @@ import {
   PickersMonthClasses,
 } from './pickersMonthClasses';
 
-interface PickersMonthProps {
+export interface ExportedPickersMonthProps {
+  classes?: Partial<PickersMonthClasses>;
+}
```

Comment:
Why export only this prop? I must admit that I do not fully understand how to choose between exported and not exported interfaces

Also, should it have a `className` like for `ExportedPickersYearProps`?

Reply:
> Why export only this prop?

This is an internal component—I'd imagine no one would want to use it standalone.
Hence, I thought, that it doesn't make sense to expose the possibility of setting other props as `defaultProps` when they are going to be overwritten by `MonthCalendar`.
Do you think that someone would want to use this component standalone? 🤔 

> Also, should it have a className like for ExportedPickersYearProps?

I'll look into addressing it. Wasn't looking into aligning everything 100% with this PR. 🙈

Reply:
In standalone probably not, But we have #4662 creating slots for month and year buttons, so dev will need to know which props they can use on those slots

Reply:
> Also, should it have a className like for ExportedPickersYearProps?

That one is tricky.
In some situations we are using the `className` prop internally so if we expose it and people override it using the theme, it will break our component.

I'm not very clear on that topic tbh.

Reply:
So I would be in favor of not exposing `className` for instance

Reply:
> In some situations we are using the className prop internally so if we expose it and people override it using the theme, it will break our component.

Are you sure it would break anything? `defaultProps` in theme definition only sets the props to defined values in case they are undefined 🤔 
If for you or another reason a higher order component is passing a `className` down - customized `defaultProps` should take no effect.

P.S. I did test the changes—both `PickersMonth` and `PickersYear` work fine with either `className` or `classes.root` defined in `defaultProps`. 🤷

Reply:
> Are you sure it would break anything? defaultProps in theme definition only sets the props to defined values in case they are undefined thinking

You are right, it would not break our code
But what would be the point of this override ?
If we take the example of the `className`, in which scenario would someone use it if it can be overridden and they don't know when ?

Reply:
Fair points.
If I follow your suggestion correctly—you are suggesting to only expose `className` to usable standalone components and not the internal/undocumented (currently) ones?

Reply:
Something like that yes
In the new pickers I have the `PickerViewLayout` component which is wrapped in a `styled` for the static pickers.
So for this component, I think that exposing `className` in the theme would make no sense for instance.

Reply:
- Removed `className` from exposed props in `PickersYear`
- Remove `className` in `PickersMonth` (it was not present—no point in adding it as long as it's not needed)